src/tests/index.tests.js

```diff
@@ -0,0 +1,407 @@
+describe ('HW_4', function() {
+    describe ('ArrayList', function() {
+        describe('Initialization (init)', function() {
+            it('should be defined', function() {
+                assert.isDefined(ArrayList.prototype.init, 'Initialization process has been defined');
+            });
+
+            it('should be Warning (undefined)', function() {
+                const arr = undefined;
+                const expSize = 'Booom try again!';
+                const expArr = 'Booom try again!';
+
+                const actual = ArrayList.prototype.init(arr);
+
+                assert.deepEqual(actual, expArr, expSize);
+            });
+
+            it('should be Warning (null)', function() {
+                const arr = null;
+                const expSize = 'Booom try again!';
+                const expArr = 'Booom try again!';
+
+                const actual = ArrayList.prototype.init(arr);
+
+                assert.deepEqual(actual, expArr, expSize);
+            });
+
+            it('should return ([1, 2, 3])', function() {
+                const array = [1, 2, 3];
+                const expArr = [1, 2, 3];
+                const expSize = 3;
+                
+
+                const actual = ArrayList.prototype.init(array);
+
+                assert.deepEqual(actual, expArr)
+                
+                assert.deepEqual(actual.length, expSize)
+            });
+        }); 
+        describe('addAtEnd (push)', function() {
+            it('should be defined', function() {
+                assert.isDefined( ArrayList.prototype.addAtEnd, 'Adding process has been defined');
+            });
+
+            it('should be Warning (undefined)', function() {
+                const arr = undefined;
+                const expSize = 'Booom try again!';
+                const expArr = 'Booom try again!';
+
+                const actual = ArrayList.prototype.addAtEnd(arr);
+
+                assert.deepEqual(actual, expArr, expSize);
```

Comment:
how the length of an array, an array and the result of invoking function could be equal the same value ?